diff --git a/build.zig b/build.zig
index 41ac5f04c..25e55aefb 100644
--- a/build.zig
+++ b/build.zig
@@ -206,17 +206,16 @@ pub fn build(b: *std.Build) !void {
         const lib_shared = try buildpkg.GhosttyLib.initShared(b, &deps);
         const lib_static = try buildpkg.GhosttyLib.initStatic(b, &deps);
 
-        // We shouldn't have this guard but we don't currently
-        // build on macOS this way ironically so we need to fix that.
-        if (!config.target.result.os.tag.isDarwin()) {
-            lib_shared.installHeader(); // Only need one header
-            if (config.target.result.os.tag == .windows) {
-                lib_shared.install("ghostty-internal.dll");
-                lib_static.install("ghostty-internal-static.lib");
-            } else {
-                lib_shared.install("ghostty-internal.so");
-                lib_static.install("ghostty-internal.a");
-            }
+        // Kodosi packages the Darwin static image into its own XCFramework.
+        lib_shared.installHeader();
+        if (config.target.result.os.tag.isDarwin()) {
+            lib_static.install("libghostty.a");
+        } else if (config.target.result.os.tag == .windows) {
+            lib_shared.install("ghostty-internal.dll");
+            lib_static.install("ghostty-internal-static.lib");
+        } else {
+            lib_shared.install("ghostty-internal.so");
+            lib_static.install("ghostty-internal.a");
         }
     }
 
